docs/documentation/test_matrix.md

```diff
@@ -7,8 +7,22 @@ Devices and configurations TrenchBoot has been tested on.
 | Intel Tiger Lake client                                                      |  TPM 2.0   |  PASS  | UEFI firmware |
 | Intel Kaby Lake server                                                       |  TPM 2.0   |  PASS  | UEFI firmware |
 | Intel Skylake sever                                                          |  TPM 2.0   |  PASS  | UEFI firmware |
+| Dell OptiPlex 9010                                                           |  TPM 1.2   |  PASS  | coreboot SeaBIOS firmware |
 | PC Engines APU2 platform series <br>(AMD family 16h models 30h-3fh embedded) |  TPM 2.0   |  PASS  | coreboot firmware |
 | PC Engines APU2 platform series <br>(AMD family 16h models 30h-3fh embedded) |  TPM 1.2   |  PASS  | coreboot firmware |
+| Asus KGPE-D16 <br>(AMD Opteron family 15h models 00h-0fh server)             |  TPM 1.2   |  PASS  | stock BIOS |
 | Asus KGPE-D16 <br>(AMD Opteron family 15h models 00h-0fh server)             |  TPM 2.0   |  FAIL  | coreboot firmware TPM issue |
 | Supermicro M11SDV-8CT <br>(AMD EPYC 3000 Snowy Owl server)                   |  TPM 2.0   |  PASS  | CSM legacy boot |
-| Supermicro M11SDV-8CT <br>(AMD EPYC 3000 Snowy Owl server)                   |  TPM 2.0   |  FAIL  | UEFI boot |
\ No newline at end of file
+| Supermicro M11SDV-8CT <br>(AMD EPYC 3000 Snowy Owl server)                   |  TPM 2.0   |  FAIL  | UEFI boot |
+
+Hardware quirks and workarounds.
+
+| Device                                                                       | Notes |
+|:----------------------------------------------------------------------------:|:-----:|
+| Dell OptiPlex 9010                                                           | 1. Installer has issues rebooting without `reboot=pci` kernel option. |
+| (continued)                                                                  | 2. Xen sometimes has issues rebooting, boot cycle is the workaround. |
+| Asus KGPE-D16 <br>(AMD Opteron family 15h models 00h-0fh server)             | IOMMU has no extended features: can't use `INVALIDATE_IOMMU_ALL` in SKL. |
+| Supermicro M11SDV-8CT <br>(AMD EPYC 3000 Snowy Owl server)                   | Problematic USB controller for Qubes OS ([resets the system][qubesos-m11-reset]). |
+| (continued)                                                                  | Works without `sys-usb` VM or if USB controller is disabled. |
```

Comment:
I think all of those should have tracking issues associated. WDYT @stacktrust @dpsmith ?

Reply:
Yes, each test failure can be linked to a tracking issue.

Current table should be separated into two tables, one for tested-good (PASS) hardware.  Add a column to that table with sales status, e.g. YYYY-YYYY (year first sold, year for End of Sale).  Currently, everything in the table is probably no longer sold.

The table with test failures can be divided into two:  hardware that can never be supported due to an unchangeable limitation, or hardware that could potentially be supported by a firmware or software (OS, driver) fix.

Somewhere on this page should be a description or link to the tests which map onto "PASS" status, e.g. is this related to fwupd hardware test version ABC? AEM-with-TrenchBoot version XYZ?

Reply:
It was added by @miczyg1 in 3a3a16d, which is why I mentioned him in another comment, the files in that import were supposedly moved from https://github.com/TrenchBoot/documentation/ repo, but I didn't find the test matrix there.

I now think that these are actual CI results related to https://github.com/3mdeb/meta-trenchboot like https://gitlab.com/trenchboot1/3mdeb/meta-trenchboot/-/pipelines/173780864.  https://github.com/3mdeb/trenchboot-ci could also be related.  This looks like Debian/yocto build with Trenchboot, not AEM/Xen/QubesOS like new test results.

Reply:
So, maybe we should explain what we mean by "tested" or "known to work" for the entries that we add now. For entries for which we don't know how testing was done or what the failure reason was, let's create an issue to investigate and find out and link that issue here so anyone who stumbles on it will poster us for an answer. Hopefully @BeataZdunczyk can allocate some resources to fix that in further phases.

I really miss transparent validation here.

Reply:
See 382a853954aa6f18db8b28e9a3b42e124b058f2f and b1c744ea2b347ff114b70cb5656b42b7b62fb863

Reply:
@SergiiDmytruk thank you, it helps a lot but is still not up to the transparent validation standard since OpenQA is vaguely mentioned.  Can we point to specific test reports maybe GitHub issues where the test results were published? So when someone would like to ask what was really tested and how, it would be answered. Also if we tested that manually and someone personally confirmed it work maybe methodology was described in some place and we can say methodology described here (link) was tested by Joe Doe at version X and he confirmed it works.

I hope younger my point. I just want to avoid similar issues we face right now with hard to identify test results. Transparent validation is the key trustworthiness ingredient.

Reply:
See dae0be6bc5b1ea2e9fbebcc8eb4c8deb43f9d064.  Test results for openQA are in the history of a non-public openQA instance.  Outcome of manual test results aren't documented in a single place, mostly comments in PRs, issues or Jira.  I've linked to openQA tests distribution that's being used, explained steps that both automatic and manual testing involves and linked to blog posts which explain the process in more detail (with commands and screenshots).

Reply:
Thank you. That resolve both conversations IMHO.